feat: explicit option to load triage file

raboof opened the pull request:
As discussed in #2316, provide a way to load the (scanned) dependency data and the (manually enriched) triage data from separate files.

Another way to implement this might have been to allow multiple `--input-file` parameters, but  making an explicit separation between dependency input and triage input seems like a good idea, as we might want to handle those differently (and not rely on detecting them by looking at file extensions) in the future.

This implementation also assumes the triage data is in `vex` format and not in the internal `json` format. I think this makes sense since you might want to exchange triage data between different parties/tools, so having the `vex` format as a 'first-class citizen' rather than supporting `vex` and the internal format side-by-side seems more attractive.

raboof commented on test/test_requirements.py:
```diff
@@ -136,7 +136,7 @@ def test_requirements():
             "--input-file",
             SCAN_CSV,
-            "--merge",
+            "--triage-input-file",
```
This did not work as the test apparently intended: because only one `--input-file` parameter is used, previously the `SCAN_CSV` was entirely ignored and the test only checked that all vulnerabilities that were present in `TRIAGE_JSON`were indeed mitigated.

Because of this, the test never noticed that we depend on `plotly.js` which has an open CVE and did not appear in the triage file.

terriko reviewed (changes requested):
Thanks for figuring this out and finding a solution.  I suspect that switching to .vex as our preferred triage probably makes the most sense beyond just this case.  When we did the original format we weren't thinking about sboms or standards compliance yet and I'm not sure we were even aware that VEX existed as a viable alternative.  

Do you mind adding `--triage-input-file` to the documentation?  There's currently a basic function list in README.md that could use an update and a bit longer explanation that would need to go in MANUAL.md.   We might want an explicit simplified test outside of our usage in test_requirements.py too, since I'm pretty test_requirements.py isn't counted in our current coverage.

raboof commented:
> I suspect that switching to .vex as our preferred triage probably makes the most sense beyond just this case. When we did the original format we weren't thinking about sboms or standards compliance yet and I'm not sure we were even aware that VEX existed as a viable alternative.

Yeah things seem to be moving fast still in this area ;)

> Do you mind adding --triage-input-file to the documentation? There's currently a basic function list in `README.md` that could use an update and a bit longer explanation that would need to go in `MANUAL.md`.

Added lines in `README.md`, `MANUAL.md` might indeed need more changes  - I can look at that later (possibly next week) but perhaps that could be a separate PR.

> We might want an explicit simplified test outside of our usage in test_requirements.py too, since I'm pretty test_requirements.py isn't counted in our current coverage.

I added an explicit `test_triage.py` that is more targeted, but I haven't looked into what would be needed to count it in our coverage. Perhaps ideally it should not use `subprocess` and test internals more directly, but on the other hand it might be tricky to trigger the problem that this PR fixes that way.

terriko commented:
re-running the failed tests now because I'm going to be stuck in a meeting for an hour anyhow, but I should have time to come back and actually do code review when my meeting is over.

terriko reviewed (approved):
Narrator: and then her meeting never ended.

Sorry about the delay.  This is looking good, though windows tests are still being weird.  I'm going to go ahead and get it merged now.  thank you!
